api/v1/stocks.go

```diff
@@ -29,24 +30,46 @@ func (api *APIHandler) getAllStocks(w http.ResponseWriter, r *http.Request) {
 
 func (api *APIHandler) getStockDetails(w http.ResponseWriter, r *http.Request) {
 	symbol := chi.URLParam(r, "symbol")
+	time := chi.URLParam(r, "time")
 
 	yamlData, err := details.StocksYamlBytes()
 	if err != nil {
 		responses.ErrorResponse(w, http.StatusInternalServerError, fetchingError.Error())
+		return
 	}
 
 	var myStocks []details.StockDetails
-
 	if err := yaml.Unmarshal(yamlData, &myStocks); err != nil {
-		responses.ErrorResponse(w, http.StatusInternalServerError, fetchingError.Error())
+		responses.ErrorResponse(w, http.StatusInternalServerError, decodingError.Error())
+		return
 	}
 
-	for i := range myStocks {
-		if myStocks[i].Symbol == symbol {
-			responses.CustomResponse(w, myStocks[i], 200)
+	stock, err := getStockHelper(myStocks, symbol)
+	if err != nil {
+		responses.ErrorResponse(w, http.StatusInternalServerError, stockNotFoundError.Error())
+		return
+	}
+	if time != "" {
```

Comment:
when time isn't specified, we should consider maybe returning the info at the current tick

Reply:
Alright, makes sense! Will change it